test/unit/test_http_utils.py

```diff
@@ -123,3 +125,30 @@ def server(client, _):
                 self.assertEqual(''.join(rest), data[off:])
         finally:
             server_thread.kill()
+
+
+# Ripoff from CPython Lib/test/test_httplib.py
+class FakeSocket(object):
+    def __init__(self, text, fileclass=StringIO.StringIO):
+        self.text = text
+        self.fileclass = fileclass
+        self.data = ''
+
+    def sendall(self, data):
+        self.data += ''.join(data)
+
+    def makefile(self, mode, bufsize=None):
+        if mode != 'r' and mode != 'rb':
+            raise httplib.UnimplementedFileMode()
+        return self.fileclass(self.text)
+
+
+class TestSomewhatBufferedHTTPConnection(unittest.TestCase):
+
+    def test_py26_reponse_instanciation(self):
```

Comment:
In this test, you might want to mock the `httplib.HTTPResponse` constructor, and assert it's called with at least all the arguments except for `buffering`, and `buffering` if the Python version supports it.